evaluation.py

```diff
@@ -315,18 +315,30 @@ def pck_eval(self, list_of_predictions):
         for prediction in list_of_predictions:
             prediction_image_id = prediction['image_id']
             prediction_keypoints = prediction['keypoints']
-            for i, ann in enumerate(data['annotations']):
+            anns = len(data['annotations'])
+            for i in range(anns):
                 if data['annotations'][i]['image_id'] == prediction_image_id:
                     annotation_keypoints = data['annotations'][i]['keypoints']
                     prediction_keypoints = np.array(prediction_keypoints)
                     annotation_keypoints = np.array(annotation_keypoints)
 
                     # Calculate PCK@0.2 threshold for image
                     # Joint at 11 is left hip, Joint at 12 is right hip. Multiply by 3 as each keypoint has (x, y, visibility) to get the array index
-                    left_hip_point = np.array(annotation_keypoints[33], annotation_keypoints[34])
-                    right_hip_point = np.array(annotation_keypoints[36], annotation_keypoints[37])
-                    torso = np.linalg.norm(left_hip_point-right_hip_point)
-                    threshold = PCK_THRESHOLD*torso
+                    if((annotation_keypoints[33] != 0 and annotation_keypoints[34] != 0) or (annotation_keypoints[36] != 0 and annotation_keypoints[37] != 0)):
```

Comment:
I think checking the visibility flag is more robust (and shorter) 
```suggestion
                    if annotation_keypoints[35] > 0 and annotation_keypoints[38] > 0:
```

Reply:
oh good call I forgot about the flag!